libmcount/dynamic.c

```diff
@@ -244,6 +244,38 @@ struct mcount_dynamic_info * create_mdi(unsigned long baseaddr,
 	return mdi;
 }
 
+struct module_list {
+	struct list_head	list;
+	char*			name;
+};
+
+static LIST_HEAD(module_list);
```

Comment:
Why is this needed?  Who frees it?

Reply:
First, I updated the code and added the code to free the module_list. and It is needed to keep a list of modules that have not yet been instrumented.

Reply:
So you did it only for dlopen, right?  I'm not sure this is necessary as we want to patch functions in dlopen library always..  What's the actual case (or problematic case) do you see?